From 2aa2a75e9974b56802662f48968c24f954f47809 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 21 Jan 2021 15:12:52 +0200
Subject: [PATCH] [SW-30593] gaudi: pll: raise sram <=> arm path plls

It was discovered that running most of the PLLs on the ARM <=> SRAM path
at their source frequency (50MHz) with no cache and MMU results in lot's
of instruction fetches from SRAM and slows down the CPU significantly,
so it even loses the data bytes on the I2C slave bus running @400KHz.
When the I2C slave bus runs @100KHz the CPU is able to get all the data
in time.
Raise the all the PLLs on the ARM <=> SRAM path to their nominal values.
We can decide later to lower them to the frequencies that will suffice,
but it will require more testing time.

Change-Id: I9616d641cf04cd912dd8438487075432aa3b06da
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/pll.c b/gaudi/pll.c
index 241433c..6230cfc 100644
--- a/gaudi/pll.c
+++ b/gaudi/pll.c
@@ -14,6 +14,69 @@
 
 static struct hl_pll_cfg gaudi_pll_cfg[] = {
 	{
+		.name = "IF_W", .base = mmIF_W_PLL_BASE & ~HL_CFG_BASE,
+		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
+		.div_factor = { 0, 1, 2, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "IF_E", .base = mmIF_E_PLL_BASE & ~HL_CFG_BASE,
+		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
+		.div_factor = { 0, 1, 2, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "MESH_W", .base = mmMESH_W_PLL_BASE & ~HL_CFG_BASE,
+		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
+		.div_factor = { 0, 1, 2, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "SRAM_W", .base = mmSRAM_W_PLL_BASE & ~HL_CFG_BASE,
+		.nr = 4, .nf = 331, .od = 3, .nb = 165, /* 830 */
+		.div_factor = { 0, -1, -1, -1 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
+		},
+	}, {
+		.name = "MESH_E", .base = mmMESH_E_PLL_BASE & ~HL_CFG_BASE,
+		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
+		.div_factor = { 0, 1, 2, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "SRAM_E", .base = mmSRAM_E_PLL_BASE & ~HL_CFG_BASE,
+		.nr = 4, .nf = 331, .od = 3, .nb = 165, /* 830 */
+		.div_factor = { 0, -1, -1, -1 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
+		},
+	}, {
+		.name = "PCI", .base = mmPSOC_PCI_PLL_BASE & ~HL_CFG_BASE,
+		.nr = 4, .nf = 331, .od = 3, .nb = 165, /* 830 */
+		.div_factor = { 0, 0, 4, 9 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
 		.name = "NIC0", .base = mmNIC0_PLL_BASE & ~HL_CFG_BASE,
 		.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
 		.div_factor = { 0, 4, 8, -1 },
